lib/fungus_toast/games/grid.ex

```diff
@@ -0,0 +1,32 @@
+defmodule FungusToast.Games.Grid do
+    alias FungusToast.Games.GridCell
+
+def create_starting_grid(number_of_rows, number_of_columns, player_ids) do
+    initial_grid_cells_map = map_of_player_id_to_random_grid_index(number_of_rows, number_of_columns, player_ids)
+
+    Enum.reduce(initial_grid_cells_map, %{}, fn {player_id, position}, map ->
+      Map.put(map, position, %GridCell{
+        index: position,
+        player_id: player_id,
+        live: true,
+        empty: false
+      })
+    end)
+  end
+
+  defp map_of_player_id_to_random_grid_index(number_of_rows, number_of_columns, player_ids) do
+    number_of_players = length(player_ids)
+    number_of_cells_to_choose_from = number_of_columns * number_of_rows / number_of_players
+    Enum.into(
```

Comment:
This whole block still feels overly clever  and I'd love to see it simplified or split into functions to at least make this easier to read. Lots of `..` for ranges and `&` that make things harder to read and understand what exactly is going on

Reply:
I think most of this will go away with predefined start positions. I'll work on this now.